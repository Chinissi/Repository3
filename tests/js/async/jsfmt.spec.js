run_spec(__dirname, ["babel", "flow", "typescript"], {
  errors: {
    flow: ["await-keyword.js"],
    espree: ["await-keyword.js"],
    meriyah: ["await-keyword.js"],
  },
});
