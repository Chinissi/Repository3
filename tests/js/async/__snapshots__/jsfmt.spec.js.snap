// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async-iteration.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================

async function * a() {
    yield* b();
}

class X {
    async * b() {
        yield* a();
    }
}

=====================================output=====================================
async function* a() {
  yield* b();
}

class X {
  async *b() {
    yield* a();
  }
}

================================================================================
`;

exports[`async-shorthand-method.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
({
  async get() {},
  async set() {}
});

=====================================output=====================================
({
  async get() {},
  async set() {},
});

================================================================================
`;

exports[`await-keyword.js [espree] format 1`] = `
"Cannot use keyword 'await' outside an async function (2:20)
  1 | async function foo() {
> 2 |   function bar(x = await 2) {}
    |                    ^
  3 | }
  4 |
  5 | async (x = await 2) => {};"
`;

exports[`await-keyword.js [flow] format 1`] = `
"Unexpected number, expected the token \`,\` (2:26)
  1 | async function foo() {
> 2 |   function bar(x = await 2) {}
    |                          ^
  3 | }
  4 |
  5 | async (x = await 2) => {};"
`;

exports[`await-keyword.js [meriyah] format 1`] = `
"[2:24]: 'Await' may not be used as an identifier in this context (2:24)
  1 | async function foo() {
> 2 |   function bar(x = await 2) {}
    |                        ^
  3 | }
  4 |
  5 | async (x = await 2) => {};"
`;

exports[`await-keyword.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function foo() {
  function bar(x = await 2) {}
}

async (x = await 2) => {};

async function foo() {
  function bar(x = await (2)) {}
}

async (x = await (2)) => {};

=====================================output=====================================
async function foo() {
  function bar(x = await 2) {}
}

async (x = await 2) => {};

async function foo() {
  function bar(x = await(2)) {}
}

async (x = await 2) => {};

================================================================================
`;

exports[`await-parse.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function f1() { (await f()).length }
async function g() {
  invariant(
    (await driver.navigator.getUrl()).substr(-7)
  );
}
function *f2(){
  !(yield a);
}
async function f3() {
  a = !await f();
}
async () => {
  new A(await x);
  obj[await x];
}

=====================================output=====================================
async function f1() {
  (await f()).length;
}
async function g() {
  invariant((await driver.navigator.getUrl()).substr(-7));
}
function* f2() {
  !(yield a);
}
async function f3() {
  a = !(await f());
}
async () => {
  new A(await x);
  obj[await x];
};

================================================================================
`;

exports[`conditional-expression.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function f() {
  const result = typeof fn === 'function' ? await fn() : null;
}

(async function() {
  console.log(
    await (true ? Promise.resolve("A") : Promise.resolve("B"))
  );
})()

async function f2() {
  await (spellcheck && spellcheck.setChecking(false));
  await spellcheck && spellcheck.setChecking(false)
}

=====================================output=====================================
async function f() {
  const result = typeof fn === "function" ? await fn() : null;
}

(async function () {
  console.log(await (true ? Promise.resolve("A") : Promise.resolve("B")));
})();

async function f2() {
  await (spellcheck && spellcheck.setChecking(false));
  (await spellcheck) && spellcheck.setChecking(false);
}

================================================================================
`;

exports[`inline-await.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function f() {
  const admins = (await(db.select('*').from('admins').leftJoin('bla').where('id', 'in', [1,2,3,4]))).map(({id, name})=>({id, name}))
}

=====================================output=====================================
async function f() {
  const admins = (
    await db
      .select("*")
      .from("admins")
      .leftJoin("bla")
      .where("id", "in", [1, 2, 3, 4])
  ).map(({ id, name }) => ({ id, name }));
}

================================================================================
`;

exports[`nested.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
const getAccountCount = async () =>
  (await
    (await (
      await focusOnSection(BOOKMARKED_PROJECTS_SECTION_NAME)
    ).findItem("My bookmarks")
  ).getChildren()
  ).length

=====================================output=====================================
const getAccountCount = async () =>
  (
    await (
      await (
        await focusOnSection(BOOKMARKED_PROJECTS_SECTION_NAME)
      ).findItem("My bookmarks")
    ).getChildren()
  ).length;

================================================================================
`;

exports[`parens.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function *f(){ await (yield x); }

async function f2(){ await (() => {}); }

=====================================output=====================================
async function* f() {
  await (yield x);
}

async function f2() {
  await (() => {});
}

================================================================================
`;

exports[`simple-nested-await.js format 1`] = `
====================================options=====================================
parsers: ["babel", "flow", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
async function f() {
  const a = await (await request()).json();
  const b = await fs.writeFile(file, await (await request()).text());
}

=====================================output=====================================
async function f() {
  const a = await (await request()).json();
  const b = await fs.writeFile(file, await (await request()).text());
}

================================================================================
`;
